src/espIdf/serial/serialPort.ts

```diff
@@ -105,4 +116,24 @@ export class SerialPort {
       }
     });
   }
+
+  private async wslList(): Promise<SerialPortDetails[]> {
+    let result = "";
+    try {
+      result = await execChildProcess(
+        `powershell.exe -Command "$(cat wsl_get_serial_list.ps)"`,
```

Comment:
is this compatible with both old wsl and wsl2 ?

Reply:
also source for this file?

Reply:
Should we add the source in the file ? I though the code is small and general enough.

The .NET Framework 2.0 defines the `System.IO.Ports.SerialPort` and should be independent of WSL version.

Will test it soon.

Reply:
I think its better to add source, in case in future we don't remember what code we wrote to generate this bin.

Also it will be much easier for git to track

Reply:
I added the source in `wsl_get_serial_list.ps`